dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/task/dependent/DependentAsyncTaskExecuteFunction.java

```diff
@@ -183,9 +192,16 @@ private List<DependentExecute> initializeDependentTaskList() {
 
     private DependResult calculateDependResult() {
         List<DependResult> dependResultList = new ArrayList<>();
+        Map<String, Long> dependVarPoolEndTimeMap = new HashMap<>();
         for (DependentExecute dependentExecute : dependentTaskList) {
             DependResult dependResult =
                     dependentExecute.getModelDependResult(dependentDate, processInstance.getTestFlag());
+            if (dependResult == DependResult.SUCCESS) {
+                Map<String, Property> varPoolPropertyMap = dependentExecute.getDependTaskVarPoolPropertyMap();
+                Map<String, Long> varPoolEndTimeMap = dependentExecute.getDependTaskVarPoolEndTimeMap();
+                dependentExecute.addTaskVarPool(varPoolPropertyMap, varPoolEndTimeMap, dependVarPoolPropertyMap,
+                        dependVarPoolEndTimeMap);
+            }
```

Comment:
No need to get two maps and then send them back as arguments, at least, first two arguments can be removed. BTW, may I ask whether local variable `dependVarPoolEndTimeMap ` in this method is useless?

Reply:
The use of `dependVarPoolEndTimeMap` is to obtain the data of `varPoolEndTimeMap` in the `addTaskVarPool` method for the comparison of the endTime of the subsequent loop

Reply:
These four parameters have different meanings in the two classes. I think it is more appropriate to regard this method as a tool method?

Reply:
> The use of dependVarPoolEndTimeMap is to obtain the data of varPoolEndTimeMap in the addTaskVarPool method for the comparison of the endTime of the subsequent loop
It seems that you just pass an empty map, maybe it's a different usage, thus, define an util class do seems more clear.

Reply:
I am also a little confused, maybe it is more appropriate to put this method in the `DependentUtils` class😂

Reply:
Thank you for helping to correct my code

Reply:
Never mind, enjoy your coding :D